src/makeRequest.ts

```diff
@@ -1,10 +1,45 @@
 import axios from 'axios'
 import * as Boom from 'boom'
+import * as jaeger from 'jaeger-client'
 import * as _ from 'lodash'
+import * as opentracing from 'opentracing'
 
-async function makeRequest (topic: string, payload: any, target: string) {
-  return axios.post(target, { topic, payload })
-    .then(response => response.data.payload)
+const config = {
+  serviceName: 'chayen',
+  sampler: {
+    type: 'const',
+    param: 1,
+    host: 'localhost',
+    port: 5775,
+    refreshIntervalMs: 500
+  },
+  reporter: {
+    flushIntervalMs: 500
+  }
+}
+
+const options = {
+  tags: {
+    service_version: '1.0.0' // could be sent from the caller?
+  }
+}
+
+async function makeRequest (topic: string, payload: any, target: string, headers?: any) {
```

Comment:
I think it is weird to allow `Chayen.makeRequest` to receive arbitrary header object. I think this kinda breaks the abstraction.

Reply:
I recommend accepting a source option to specify the source of first request (which is then added as a tag):

```js
  const res = await Chayen.makeRequest(
    'test:delegate:plus3',
    { number: 2 },
    `http://localhost:${server.getAddress().port}/rpc`,
    { source: 'test/integration/endpointDelegate' }
  )
```

In `makeDelegateRequestAsync`, instead of passing a header, you can make it pass a parentSpan:

```js
          return makeRequest(topic, payload, target, { parentSpan: span })
```